Tests/MapboxMapsTests/Foundation/DictionaryEncoderTests.swift

```diff
@@ -0,0 +1,436 @@
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<Int64>(value: nil)) as? [String: Int64?],
+            ["value": nil]
+        )
+
+        // UInt
+        XCTAssertEqual(
+            try sut.encode(DummyNilable(value: UInt8(10))) as? [String: UInt8?],
+            ["value": 10]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<UInt8>(value: nil)) as? [String: UInt8?],
+            ["value": nil]
+        )
+
+        XCTAssertEqual(
+            try sut.encode(DummyNilable(value: UInt16(10))) as? [String: UInt16?],
+            ["value": 10]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<UInt16>(value: nil)) as? [String: UInt16?],
+            ["value": nil]
+        )
+
+        XCTAssertEqual(
+            try sut.encode(DummyNilable(value: UInt32(10))) as? [String: UInt32?],
+            ["value": 10]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<UInt32>(value: nil)) as? [String: UInt32?],
+            ["value": nil]
+        )
+
+        XCTAssertEqual(
+            try sut.encode(DummyNilable(value: UInt64(10))) as? [String: UInt64?],
+            ["value": 10]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<UInt64>(value: nil)) as? [String: UInt64?],
+            ["value": nil]
+        )
+
+        // Boolean
+        XCTAssertEqual(
+            try sut.encode(DummyNilable(value: false)) as? [String: Bool?],
+            ["value": false]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<Bool>(value: nil)) as? [String: Bool?],
+            ["value": nil]
+        )
+    }
+
+    func testEncodeNilOnlyIfPresent() throws {
+        let sut = DictionaryEncoder()
+        sut.userInfo[.shouldEncodeNilValues] = false
+
+        // String
+        XCTAssertEqual(
+            try sut.encode(DummyNilable(value: "hilla")) as? [String: String?],
+            ["value": "hilla"]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<String>(value: nil)) as? [String: String?],
+            [:]
+        )
+
+        // Double
+        XCTAssertEqual(
+            try sut.encode(DummyNilable(value: 10.0)) as? [String: Double?],
+            ["value": 10.0]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<Double>(value: nil)) as? [String: Double?],
+            [:]
+        )
+
+        // Float
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<Float>(value: 10.0)) as? [String: Float?],
+            ["value": 10.0]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<Float>(value: nil)) as? [String: Float?],
+            [:]
+        )
+
+        // Int
+        XCTAssertEqual(
+            try sut.encode(DummyNilable(value: 10)) as? [String: Int?],
+            ["value": 10]
+        )
+        XCTAssertEqual(
+            try sut.encode(DummyNilable<Int>(value: nil)) as? [String: Int?],
```

Comment:
There's some funky stuff going on - this works with a generic struct. But when using a similar specialized struct - the result is an empty dictionary.

```swift
    private struct DummyFloatEncodable: Encodable {
        let value: Float?
    }
```

Reply:
Maybe a sure way to test these would be to define a dedicated struct for every test 🤔

Reply:
yes, I am also thinking of using stylegen to generate the tests case for each Layer's struct :-D let's see!

Reply:
This test is expected the dictionary to be empty though? Did you mean to comment this in other places?